lib/pf/nodecategory.pm

```diff
@@ -81,9 +81,23 @@ sub nodecategory_populate_from_config {
         $logger->error($msg);
         return;
     }
+    my @keep;
     while(my ($id, $role) = each(%$config)) {
+        push @keep, $id;
         nodecategory_upsert($id, %$role);
     }
+    _nodecategory_bulk_delete(\@keep);
+}
+
+sub _nodecategory_bulk_delete {
+    my ($keep) = @_;
+    pf::dal::nodecategory->remove_items(
```

Comment:
This has the potential to fail due to constaints I believe.

I can imagine someone deleting the role from the configuration file manually and reloading the config.

In that case it wouldn't output the fact that the role can't be deleted from the DB due to constaints and that will have the logs show an SQL error on every configreload hard

To prevent that, the configreload hard output should say if this succeeded or not